### ISSUE TITLE
Updating the PUF

### ISSUE THREAD

**Issue by andersonfrailey** [open-source-economics/taxdata#38]
TaxCalc currently uses a PUF file made from a statistical match of the [2014 CPS](http://www.nber.org/data/current-population-survey-data.html) and [2009 Public Use Tax File](http://users.nber.org/~taxsim/gdb/gdb09.pdf) from the IRS. I am currently working on updating the PUF with new variables, new weights, and the 2015 CPS/2010 PUF in the following steps:
1. Add dependent ages from the CPS to the PUF for use in [TaxCalc PR #976](https://github.com/open-source-economics/Tax-Calculator/pull/976) and remove p87482 (PR #37 from @martinholmer). This should cause no change in the TaxCalc output. However, I'm having problems creating an exact replica of our current PUF, which is resulting in slightly different outputs. This will be further addressed in a forthcoming TaxData PR.
2. Add targets for the number of filers with AGI above $5 million to the SOI estimates used during Stage 1 of the data extrapolation to get a more accurate distribution of income.
3. Update the CBO forecasts used in Stage 1 of the data extrapolation for more accurate growth rates.

(Steps 2 and 3 will result in more accurate weights for each of the records.)
1. Use the 2010 PUF and 2015 CPS datasets to create the match. 

The reason for doing this step-by-step rather than as one big project is to document how each change affects TaxCalc's outputs. 

I'd love any feedback on this plan or other opinions on what the next steps should be.

@martinholmer @Amy-Xu @talumbau @GoFroggyRun @MattHJensen

**Comment by MattHJensen** [comment]
@andersonfrailey, does (2) above require doing a new CPS-PUF match, or could (2) be added to the current CPS-PUF match file and thereby sidestep the issues you are having with https://github.com/open-source-economics/taxdata/pull/39.

**Comment by andersonfrailey** [comment]
@MattHJensen I would want to confirm with @Amy-Xu, but unless I'm missing a step running the current CPS-PUF match file through the extrapolation process with the new targets in (2) shouldn't be an issue.

**Comment by Amy-Xu** [comment]
It seems to me @andersonfrailey has already included 1-3 in #39? Is that right?

Also for both 2 & 3, Anderson do you mean modifying Stage I script to update stage II targets? Or I missed something here?

**Comment by andersonfrailey** [comment]
@Amy-Xu I only included (1) in #39. 

> Also for both 2 & 3, Anderson do you mean modifying Stage I script to update stage II targets? Or I missed something here?

Yes. Updating the SOI estimates CSV to include the $5 million threshold and the IRS data up to 2014 and CBO baseline to account for their new projections to update stage II targets.

**Comment by martinholmer** [comment]
@andersonfrailey, What is the status of taxdata issue #38?  Seems like issue #38 combines quite a few steps into one issue.  Haven't some of the steps mentioned in #38 already been taken?  And doesn't your plan for adding new Stage3 AGI-strata adjustment ratios make other steps obsolete?

@MattHJensen @Amy-Xu

**Comment by andersonfrailey** [comment]
@martinholmer issue #38 is mostly complete. I've added the variables discussed, we've decided against adding a $5 million target in favor of adjustment factors, and I've completed the CBO data updates.

All that is left that was mentioned in this PR is creating the new `puf.csv` file from the 2010 PUF and 2016 CPS (original post stated 2015, but in the time that has passed the 2016 is now available), and I will begin that process after completing my work on stage three.
